cmd/controller-manager/app/serve.go

```diff
@@ -33,11 +32,23 @@ import (
 	"k8s.io/kubernetes/pkg/util/configz"
 )
 
-type serveFunc func(handler http.Handler, shutdownTimeout time.Duration, stopCh <-chan struct{}) error
+// DefaultBuildHandlerChain takes in handler and CompletedConfig and returns a handler.
+func DefaultBuildHandlerChain(apiHandler http.Handler, c *CompletedConfig) http.Handler {
```

Comment:
nit: as this is hardcoded anyway, we could just call is `BuildHandlerChain`. In the apiserver it can be customized for the config. So we called it `DefaultBuildHandlerChain` because it is the default func.

Reply:
oh, ic, missed it.  fixed now